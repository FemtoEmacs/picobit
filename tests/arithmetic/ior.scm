;; bitwise or tests

(define-macro (test-cond condition name) ;; TODO watch out for name capture
  `(let ((DUMMY-x ,condition))
     (display ,name)
     (if DUMMY-x
	 (display " : PASSED\n")
	 (begin (display " : FAILED, got : ")
		(display DUMMY-x)
		(display "\n")))))

(define-macro (test-= lhs rhs name) ;; TODO watch out for name capture
  `(let ((DUMMY-lhs ,lhs)
	 (DUMMY-rhs ,rhs))
     (display ,name)
     (if (= DUMMY-lhs DUMMY-rhs)
	 (display " : PASSED\n")
	 (begin (display " : FAILED, got : ")
		(display DUMMY-lhs)
		(display " and ")
		(display DUMMY-rhs)
		(display "\n")))))

(test-= (bitwise-ior #xf0 #x0f) #xff "#xf0 | #x0f = #xff")
(test-= (bitwise-ior #x30 #x00) #x30 "#x30 | #x00 = #x30")
(test-= (bitwise-ior #x00 #x05) #x05 "#x00 | #x05 = #x05")
(test-= (bitwise-ior #x08 #x05) #x0D "#x08 | #x05 = #x0D")
(test-= (bitwise-ior #x18 #x05) #x1D "#x18 | #x05 = #x1D")
(test-= (bitwise-ior #x18 #x25) #x3D "#x18 | #x25 = #x3D")
(test-= (bitwise-ior #x18 #x35) #x3D "#x18 | #x35 = #x3D")
(test-= (bitwise-ior #x1823122312 #x351234123456) 58386709362518
	"#x1823122312 | #x351234123456 = 58386709362518")
