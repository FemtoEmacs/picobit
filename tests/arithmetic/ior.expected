:0080000080
:10800000FBD7F00281834000818C40008125400035
:108010008122400081884000018A2312819440001F
:108020008199400081A3400001C2345681C44000C0
:1080300081D8400081DC400081E0400081E34000C5
:1080400081EB400081634000016F3756813540006D
:1080500081494000803D018480340185803601865D
:1080600080240187802401898037018B8048018C1E
:108070008027018D8035018E8024018F802401901E
:10808000807C0191807C019280240193802701955E
:10809000803C019680400197804101988048000211
:1080A0008040019A807C019B8024019C807C019D02
:1080B0008048019E8034019F806801A7803901A01B
:1080C000802401A1803601A2802701A4807C01A523
:1080D00000043512802401A6803201A7802401A863
:1080E000803901A9807C01AA804501AB808001AC68
:1080F000808001AD804501AE803401AF803501B094
:10810000802701B1806901B2803701B3803C01B49E
:10811000803501B5806601B6802701B7803C01B883
:10812000808001B9807301BA802401BB806501BCE5
:10813000803A01BD802401BE803701BF807C01C030
:1081400080780002802401C1802401C3803401C5ED
:10815000803701C6802401C7802401C8803401C94A
:10816000803C01CA802401CB807C01CC803701CDCA
:10817000808001CE806601CF802701D0807C01D134
:10818000802701D2807D01D380390002803C01D458
:10819000803C01D5802401D6802401D7803C000298
:1081A000807C01D9808001DA807C01DB803401DDB4
:1081B00001DE3712807C01DF802401E1802401E2AE
:1081C000805001E4802701E5802701E6805701E720
:1081D000808001E8803701E9803401EA807801EE8F
:1081E000803701EC802401ED807001EE803701EFD3
:1081F000803B01F0802701F1802401F2807C01F3B3
:108200008024014E8037014F803901508036015162
:10821000802701528027015380800154001C2312C3
:1082200080390155800E0002807C0156803601574E
:1082300080410158803E0159806A015A8035015BB6
:10824000803E015C8027015D807C015E8024015FAF
:10825000807301608024016180270162807701645E
:10826000803401658039016680340167803001689F
:10827000806A01698035016A8027016B8024016C66
:108280008027016D806A00028034016E80270170B2
:108290008024000280410171803A01728035012AF8
:1082A000804D01738024017480240175805701760C
:1082B0008024017780360178807C01798067017A9B
:1082C000803400028036017B803C017C8073017D1C
:1082D000807C017E803A017F80240180807801814A
:1082E0008027018280720131803601328041013362
:1082F000803501348036018B8027013601373412F6
:1083000080410138802401398039013A8038013BAD
:108310008027013C804A013D8024013E8024013FAA
:10832000805401408034014180390142802401435E
:10833000806E0144807C014580350146807C014788
:10834000807101488037014A8024014B806B014CC9
:108350008041014D802701248034012A8024012698
:108360008035012780270128802401290004351A3F
:108370008037012A8027012B8027012C8041012D85
:108380008027012E8049012F807C0130807C011DD7
:108390008049011E8024011F80380120803901217D
:1083A000802701238035011A8080011B8042000252
:1083B0008034011C80340118807C0119803E01A7A3
:1083C000806A01A402B406535002B4064D51A0F4D1
:1083D00013F5A10302B4062262EEA03404F5A03422
:1083E00002B405F762EE0409F50902B405CC62EEA9
:1083F0000C09F51102B405A162EE1C09F5A02102D9
:10840000B4057662EE1CA029F5A04102B4054B62CA
:10841000EE1CA039F5A04102B4052062EEA109A12D
:108420000DF5A11502B4042B62EED802A10BB58B99
:10843000EE2022C7B30505A106B6800120DAB89563
:108440004121DBB6800221CC9A21CE2161EE20228F
:10845000CFB67200EF20B6800120DA9EA02607E694
:1084600020B558EEA02607E600EF20C0B8BF20B523
:1084700082B648012004C8B8AEA03121B5A402B527
:1084800083CDDCEF02A034210EC5C122CD2102CD67
:10849000B404947101210EC89220EF210E40622194
:1084A000B662010421C2EF2002B559B75520CCB8FD
:1084B000B8A02C07E620CEB51FEE20CFB680012055
:1084C000D296A02D07E600EF20CC9DA02407E62041
:1084D000CEB505EE20CFB666A10EB0043BEE20B0BF
:1084E0000458EEA02D07E600EF20D995A111B104A4
:1084F0003B20F09C2095A108B1043BA116B1043BA0
:10850000A110B1043BA107B0043BEE20B0043BEE48
:10851000A104B0043BEE21B0043BEEA105B1043B45
:1085200002A10FB0043BEE2022C795A106B1043B87
:10853000A107B0043BEE20B0043BEEA104B0043B25
:10854000EE21B0043BEEA105B1043B02A114B0043E
:108550003BEE2022C795A106B1043BA107B0043B26
:10856000EE20B0043BEEA104B0043BEE21B0043B8E
:10857000EEA105B1043B02A10CB0043BEE2022C7E2
:1085800095A106B1043BA107B0043BEE20B0043B2B
:10859000EEA104B0043BEE21B0043BEEA105B10412
:1085A0003B02A112B0043BEE2022C795A106B10404
:1085B0003BA107B0043BEE20B0043BEEA104B004A5
:1085C0003BEE21B0043BEEA105B1043B02A10AB091
:1085D000043BEE2022C795A106B1043BA107B004DD
:1085E0003BEE20B0043BEEA104B0043BEE21B0040E
:1085F0003BEEA105B1043B02A117B0043BEE2022E3
:10860000C795A106B1043BA107B0043BEE20B0041E
:108610003BEEA104B0043BEE21B0043BEEA105B15A
:10862000043B02A113B0043BEE2022C795A106B182
:10863000043BA107B0043BEE20B0043BEEA104B024
:10864000043BEE21B0043BEEA105B1043B01200741
:10865000E600EF022004C8212302CDCDB4066071EC
:10866000012204C822242302CDCDCDB4066F7101AE
:1086700023B8DC21B004A223222602CDCDCDB4063E
:1086800082710122B8C621B004A221242602CDCDD8
:10869000CDB4069571012022242602CDCDCDCDB4D6
:1086A00006A370002121C4232502CDCDB406B071EC
:1086B0000122B893219220EF2298019420B104A2C4
:1086C00020EF21B87BB77500B86EB76A21B73B21A0
:0286D000B725CC
:00000001FF
